pkg/sql/stats/automatic_stats.go

```diff
@@ -326,7 +326,7 @@ func (r *Refresher) ensureAllTables(
 	getAllTablesQuery := fmt.Sprintf(
 		`
 SELECT table_id FROM crdb_internal.tables AS OF SYSTEM TIME '-%s'
-WHERE schema_name = 'public'
+WHERE database_name IS NOT NULL
```

Comment:
Is it a problem for this to return tables in virtual schemas?

Reply:
virtual schemas have a null database in this table.